Data/Vector/Generic.hs

```diff
@@ -1334,6 +1334,14 @@ filterM :: (Monad m, Vector v a) => (a -> m Bool) -> v a -> m (v a)
 {-# INLINE filterM #-}
 filterM f = unstreamM . Bundle.filterM f . stream
 
+mapMaybeM :: (Monad m, Vector v a, Vector v b) => (a -> m (Maybe b)) -> v a -> m (v b)
+{-# INLINE mapMaybeM #-}
+mapMaybeM f = unstreamM . Bundle.mapMaybeM f . stream
+
+imapMaybeM :: (Monad m, Vector v a, Vector v b)
```

Comment:
Is it worthwhile to mark `imapMaybeM` as `INLINE`? (Genuine question—I'm not sure if it would be beneficial here or not.)

Reply:
I never use `uncurry` because it introduces laziness I virtually never need. I bet the compiler often fixes that, but I see no reason to tempt fate.

Reply:
I sure hope that helps. @bgamari has pointed out some trouble fusing `unstreamM`, but we surely want to fuse on the `stream` side.